botnim/query.py

```diff
@@ -111,20 +112,115 @@ def get_index_mapping(self) -> Dict:
+            context_name (str, optional): Name of the context to search in (defaults to self.context_name)
+            num_results (int, optional): Number of results to return, or None to use context default
+        
+        Returns:
+            List[SearchResult]: List of search results
+        """
+        # Use instance values if not provided
+        bot_name = bot_name or self.bot_name
+        context_name = context_name or self.context_name
+        
+        if num_results is None:
+            # Use the method to get default num_results
+            num_results = self.get_default_num_results(context_name)
+        
+        return self.search(query_text, num_results)
+
+def run_query(environment: str, bot_name: str, context_name: str, query: str, num_results: int = None, format: str = "text"):
     """
     Run a query against the vector store
     
     Args:
-        query_text (str): The text to search for
+        environment (str): Environment to use (production or staging)
         bot_name (str): Name of the bot to use
-        num_results (int): Number of results to return
+        context_name (str): Name of the context to search in
+        query (str): The search query text
+        num_results (int, optional): Number of results to return, or None to use context default
+        format (str, optional): Output format - "dict" or "text"
         
     Returns:
-        List[SearchResult]: List of search results
+        Union[List[Dict], str]: Search results in the requested format
     """
-    client = QueryClient(environment, bot_name, context_name)
-    return client.search(query_text, num_results)
+    try:
+        logger.info(f"Running vector search with query: {query}, bot: {bot_name}, context: {context_name}, num_results: {num_results}")
+        
+        client = QueryClient(environment, bot_name, context_name)
+        results = client.query(query_text=query, num_results=num_results)
+        
+        # Log the results
+        logger.info(f"Search results: {results}")
+        
+        # Format results if requested
+        if format == "text":
+            formatted_results = format_search_results(results)
+            logger.info(f"Formatted results: {formatted_results}")
+            return formatted_results
+        return results
+    except Exception as e:
+        logger.error(f"Error in run_query: {str(e)}")
+        # Return a meaningful error message instead of raising
+        return f"Error performing search: {str(e)}"
+
+def format_search_results(results: List[SearchResult], format_type: str = 'text') -> Union[str, List[Dict]]:
+    """
+    Format search results in different output formats
+    
+    Args:
+        results (List[SearchResult]): The search results to format
+        format_type (str): The output format type ('text' or 'dict')
+        
+    Returns:
+        Union[str, List[Dict]]: Formatted search results in the requested format
+    """
+    if format_type == 'dict':
```

Comment:
when is this branch used?

Reply:
never, and only the git god knows how you saw it...
fixed.